File: task_equivalents/llm_translations/llama/optimized/python/machine_learning_workflows/01_pipeline_cv_gridsearch_classification.py
import pandas as pd
from sklearn import svm
from sklearn.model_selection import GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline

# Load data
auto = pd.read_csv('data/csv/Auto.csv', na_values='?', index_col=0).dropna()

# Preprocess data
auto['mpg01'] = (auto['mpg'] > auto['mpg'].median()).astype(int)
auto = auto.drop(['mpg', 'name'], axis=1)

var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']

# Split data into features and target
x = auto[var_to_scale + var_no_scale[:-1]]
y = auto['mpg01'].astype('category').cat.codes

# Define pipelines and hyperparameter tuning spaces
ctry = [0.01, 0.1, 1, 10, 100, 1000]
tune_ctrl = {'cv': 5}

pipelines = {
    'linear': Pipeline([('scaler', StandardScaler()), ('svm', svm.SVC(kernel='linear'))]),
    'polynomial': Pipeline([('scaler', StandardScaler()), ('svm', svm.SVC(kernel='poly'))]),
    'radial': Pipeline([('scaler', StandardScaler()), ('svm', svm.SVC(kernel='rbf'))])
}

param_grids = {
    'linear': {'svm__C': ctry},
    'polynomial': {'svm__C': ctry, 'svm__degree': [2, 3, 4, 5]},
    'radial': {'svm__C': ctry, 'svm__gamma': [0.5, 1, 2, 3]}
}

# Perform hyperparameter tuning
for kernel, pipeline in pipelines.items():
    grid_search = GridSearchCV(pipeline, param_grids[kernel], **tune_ctrl, random_state=1)
    grid_search.fit(x, y)
    print(f"Kernel: {kernel}")
    print(grid_search.cv_results_)
    print(f"Best Parameters: {grid_search.best_params_}")
    print(f"Best Score: {grid_search.best_score_}")
    print()

File: task_equivalents/llm_translations/llama/optimized/r/machine_learning_workflows/01_pipeline_cv_gridsearch_classification.R
# Install required packages if not already installed
# install.packages(c("readr", "dplyr", "caret", "kernlab"))

# Load necessary libraries
library(readr)
library(dplyr)
library(caret)
library(kernlab)

# Load data
Auto <- read_csv('data/csv/Auto.csv', na = c("?", "NA")) %>%
  drop_na()

# Create binary mpg variable
Auto$mpg01 <- ifelse(Auto$mpg > median(Auto$mpg), 1, 0)
Auto <- Auto %>% 
  select(-c(mpg, name))

# Scale variables
var_no_scale <- c('cylinders', 'year', 'origin', 'mpg01')
var_to_scale <- c('displacement', 'horsepower', 'weight', 'acceleration')
preProc <- preProcess(Auto[, var_to_scale], method = c("center", "scale"))
df <- data.frame(predict(preProc, Auto[, var_to_scale]), 
                 Auto[, var_no_scale])

# Split data into features (x) and target (y)
x <- df %>% select(-mpg01)
y <- df$mpg01

# Define training control
train_control <- trainControl(method = "cv", 
                               number = 5, 
                               savePredictions = "final", 
                               classProbs = TRUE)

# Tune linear SVM model
tune_grid_linear <- expand.grid(C = c(0.01, 0.1, 1, 10, 100, 1000))
svm_linear <- train(x, y, 
                     method = "svmLinear", 
                     tuneGrid = tune_grid_linear, 
                     trControl = train_control)

# Tune polynomial SVM model
tune_grid_poly <- expand.grid(C = c(0.01, 0.1, 1, 10, 100, 1000), 
                              degree = c(2, 3, 4, 5), 
                              scale = 1)
svm_poly <- train(x, y, 
                  method = "svmPoly", 
                  tuneGrid = tune_grid_poly, 
                  trControl = train_control)

# Tune radial SVM model
tune_grid_rbf <- expand.grid(sigma = c(0.5, 1, 2, 3), 
                             C = c(0.01, 0.1, 1, 10, 100, 1000))
svm_rbf <- train(x, y, 
                 method = "svmRadial", 
                 tuneGrid = tune_grid_rbf, 
                 trControl = train_control)